src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -692,6 +692,10 @@ public synchronized void rollback() throws DatabaseEngineRuntimeException {
          * Reconnect on this method does not make sense since a new connection will have nothing to rollback.
          */
         try {
+            // While rolling back, the entities can still have batch data which needs to be cleared.
+            for (final MappedEntity mappedEntity : entities.values()) {
+                mappedEntity.getInsert().clearBatch();
```

Comment:
Shouldn't this also be done for 'getInsertWithAutoInc()' and 'getInsertReturning()' ?

Reply:
Yes, you're right. I've updated the pull request.